### PR TITLE
Improvement: Add a way to get timezone from environment.

### PR DESCRIPTION
**Pull request by dev-ddoe**
Firstly get timezone from environment. If not present, use file method

### CODE REVIEW

**Review by akallabeth — CHANGES_REQUESTED** [review]

File: winpr/libwinpr/timezone/timezone.c [hunk 1]
```diff
@@ -283,9 +283,16 @@ static BOOL winpr_match_unix_timezone_identifier_with_list(const char* tzid, con
 static TIME_ZONE_ENTRY* winpr_detect_windows_time_zone(void)
 {
 	size_t i, j;
-	char* tzid;
+	char* tzid = NULL;
+
+	char* tzenv = getenv("TZ");
+	if (tzenv) {
```
[the remark this answers is not in the document]
Comment (dev-ddoe) [reply winpr/libwinpr/timezone/timezone.c:289]: good point. I will fix that.

File: winpr/libwinpr/timezone/timezone.c [hunk 2]
```diff
@@ -283,9 +283,16 @@ static BOOL winpr_match_unix_timezone_identifier_with_list(const char* tzid, con
 static TIME_ZONE_ENTRY* winpr_detect_windows_time_zone(void)
 {
 	size_t i, j;
-	char* tzid;
+	char* tzid = NULL;
+
+	char* tzenv = getenv("TZ");
```
[the remark this answers is not in the document]
Comment (dev-ddoe) [reply winpr/libwinpr/timezone/timezone.c:288]: good point. I will fix that.

**Comment by akallabeth** [comment]
@freerdp-bot test

File: winpr/libwinpr/timezone/timezone.c [hunk 3]
```diff
@@ -284,15 +285,19 @@ static TIME_ZONE_ENTRY* winpr_detect_windows_time_zone(void)
 {
 	size_t i, j;
 	char* tzid = NULL;
+	LPCSTR tz = "TZ";
 
-	char* tzenv = getenv("TZ");
-	if (tzenv) {
-		tzid = strdup(tzenv);
+	DWORD nSize = GetEnvironmentVariableA(tz, NULL, 0);
+	if (nSize) {
```
Comment (akallabeth) [inline winpr/libwinpr/timezone/timezone.c:291]: style wrong, run `clang-format` on your changes.

Comment (dev-ddoe) [reply winpr/libwinpr/timezone/timezone.c]: Updated

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]

**Pull request merged**